drivers/auxiliary/pegasus_uch.cpp

```diff
@@ -0,0 +1,525 @@
+            RebootSP.s = reboot() ? IPS_OK : IPS_ALERT;
+            IDSetSwitch(&RebootSP, nullptr);
+            LOG_INFO("Rebooting device...");
+            return true;
+        }
+
+
+
+        // Power LED
+        if (!strcmp(name, PowerLEDSP.name))
+        {
+            int prevIndex = IUFindOnSwitchIndex(&PowerLEDSP);
+            IUUpdateSwitch(&PowerLEDSP, states, names, n);
+            if (setPowerLEDEnabled(PowerLEDS[0].s == ISS_ON))
+            {
+                PowerLEDSP.s = IPS_OK;
+            }
+            else
+            {
+                IUResetSwitch(&PowerLEDSP);
+                PowerLEDS[prevIndex].s = ISS_ON;
+                PowerLEDSP.s = IPS_ALERT;
+            }
+
+            IDSetSwitch(&PowerLEDSP, nullptr);
+            return true;
+        }
+
+        if (!strcmp(name, USBPort1SP.name))
+        {
+            return  this->setUSBPort(1, USBPort1S, USBPort1SP, states, names, n);
+        }
+        else if(!strcmp(name, USBPort2SP.name))
+        {
+            return  this->setUSBPort(2, USBPort1S, USBPort1SP, states, names, n);
+        }
+        else if(!strcmp(name, USBPort3SP.name))
+        {
+            return  this->setUSBPort(3, USBPort1S, USBPort1SP, states, names, n);
+        }
+        else if(!strcmp(name, USBPort4SP.name))
+        {
+            return  this->setUSBPort(4, USBPort1S, USBPort1SP, states, names, n);
+        }
+        else if(!strcmp(name, USBPort5SP.name))
+        {
+            return  this->setUSBPort(5, USBPort1S, USBPort1SP, states, names, n);
+        }
+        else if(!strcmp(name, USBPort6SP.name))
+        {
+            return  this->setUSBPort(6, USBPort1S, USBPort1SP, states, names, n);
+        }
+    }
+
+    return DefaultDevice::ISNewSwitch(dev, name, states, names, n);
+}
+
+
+bool PegasusUCH::setUSBPort(uint8_t port, ISwitch usbPortS[2], ISwitchVectorProperty sp, ISState * states, char * names[],
+                            int n)
+{
+    int prevIndex = IUFindOnSwitchIndex(&sp);
+    IUUpdateSwitch(&sp, states, names, n);
+    if (setUSBPortEnabled(port, usbPortS[1].s == ISS_ON))
+    {
+        sp.s = IPS_OK;
+    }
+    else
+    {
+        IUResetSwitch(&sp);
+        usbPortS[prevIndex].s = ISS_ON;
+        sp.s = IPS_ALERT;
+    }
+
+    IDSetSwitch(&sp, nullptr);
+    return true;
+}
+
+
+//////////////////////////////////////////////////////////////////////
+///
+//////////////////////////////////////////////////////////////////////
+bool PegasusUCH::ISNewNumber(const char * dev, const char * name, double values[], char * names[], int n)
+{
+    if (dev && !strcmp(dev, getDeviceName()))
+    {
+
+
+    }
+    return INDI::DefaultDevice::ISNewNumber(dev, name, values, names, n);
+}
+
+
+//////////////////////////////////////////////////////////////////////
+///
+//////////////////////////////////////////////////////////////////////
+bool PegasusUCH::ISNewText(const char * dev, const char * name, char * texts[], char * names[], int n)
```

Comment:
No need to implement this function if it is not doing anything. Unless you plan to add stuff here in the future?